### PR TITLE
Events interface: commander (w/o preflight checks) + navigator

### PR DESCRIPTION
**Pull request by bkueng**
Follow-up to https://github.com/PX4/PX4-Autopilot/pull/17479

There's room for further improvements, but should be a good baseline.

### CODE REVIEW

**Comment by dagar** [comment]
Nearly 4kB over flash on fmu-v2. We could probably ignore it temporarily, the px4io lobotomy will save more than 20 kB.

**Comment by bkueng** [comment]
Yeah worse than I thought. Either that or we could strip out MC or FW completely.

**Review by julianoes — COMMENTED** [review]
Why does it require a `\t` at the end of each `mavlink_log_...`?

**Comment by bkueng** [comment]
See description for https://github.com/PX4/PX4-Autopilot/pull/17479 (for compatibility). They will be removed later.

**Comment by dagar** [comment]
There should be enough flash now.

**Comment by bkueng** [comment]
Getting closer:
```
Memory region         Used Size  Region Size  %age Used
           flash:     1033545 B      1008 KB    100.13%
            sram:       25492 B       192 KB     12.97%
          ccsram:          0 GB        64 KB      0.00/opt/gcc/bin/../lib/gcc/arm-none-eabi/9.3.1/../../../../arm-none-eabi/bin/ld: px4_fmu-v2_default.elf section `.data' will not fit in region `flash'
/opt/gcc/bin/../lib/gcc/arm-none-eabi/9.3.1/../../../../arm-none-eabi/bin/ld: region `flash' overflowed by 1353 bytes
collect2: error: ld returned 1 exit status
```

File: src/modules/commander/Commander.cpp [hunk 1]
```diff
@@ -565,7 +616,9 @@ transition_result_t Commander::disarm(arm_disarm_reason_t calling_reason)
 					 hrt_elapsed_time(&_boot_timestamp), calling_reason);
 
 	if (arming_res == TRANSITION_CHANGED) {
-		mavlink_log_info(&_mavlink_log_pub, "Disarmed by %s", arm_disarm_reason_str(calling_reason));
+		mavlink_log_info(&_mavlink_log_pub, "Disarmed by %s\t", arm_disarm_reason_str(calling_reason));
+		events::send<events::px4::enums::arm_disarm_reason_t>(events::ID("commander_disarmed_by"), events::Log::Info,
+				"Disrmed by {1}", calling_reason);
```
Comment (cmic0) [inline src/modules/commander/Commander.cpp:621]: Typo `Disrmed` -> `Disarmed`

**Comment by bkueng** [comment]
CI is finally passing now.

Everything should be converted now, except for the LPE module.

**Comment by dagar** [comment]
> CI is finally passing now.
> 
> Everything should be converted now, except for the LPE module.

Fully passing now, the cube orange on the test rack was temporarily unavailable.

**Review by dagar — APPROVED** [review]

**Pull request merged**